crates/turborepo-lib/src/run/summary/execution.rs

```diff
@@ -248,41 +250,24 @@ enum Event {
     BuildFailed,
     Cached,
     Built,
-}
-
-#[derive(Debug, Serialize, Clone)]
-pub enum ExecutionState {
+    // Canceled due to external signal or internal failure
     Canceled,
-    Built { exit_code: i32 },
-    Cached,
-    BuildFailed { exit_code: i32, err: String },
-    SpawnFailed { err: String },
-}
-
-impl ExecutionState {
-    pub fn exit_code(&self) -> Option<i32> {
-        match self {
-            ExecutionState::BuildFailed { exit_code, .. } | ExecutionState::Built { exit_code } => {
-                Some(*exit_code)
-            }
-            _ => None,
-        }
-    }
 }
 
 #[derive(Debug, Serialize, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct TaskExecutionSummary {
-    start_time: i64,
-    end_time: i64,
-    #[serde(skip)]
-    state: ExecutionState,
-    exit_code: Option<i32>,
+    pub start_time: i64,
+    pub end_time: i64,
+    pub exit_code: Option<i32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error: Option<String>,
 }
 
 impl TaskExecutionSummary {
     pub fn is_failure(&self) -> bool {
-        matches!(self.state, ExecutionState::BuildFailed { .. })
+        // TODO: how do we treat a "none"
+        !matches!(self.exit_code, Some(0))
```

Comment:
This changes better matches the Go behavior as spawn failures/external killings are considered failures, not just non-zero exit codes.